sdk/highlight-ruby/highlight/test/highlight_test.rb

```diff
@@ -0,0 +1,30 @@
+require_relative "./test_helper"
+
+class HighlightTest < Minitest::Test
+  def test_logger
+    Highlight::H.new("qe9y4yg1")
```

Comment:
Could `logger` have been a param of the initializer? Maybe that would have avoided the singleton?
For example:

```rb
logger = Highlight::Logger.new(STDOUT)
Highlight::H.new("qe9y4yg1", logger)
```

`logger` could be an optional parameter meaning that if one is not provided then the end user doesn't want logging enabled.

Reply:
is the idea here just so the `logger` is accessible as `Highlight::H.instance.logger` or similar?

Reply:
> is the idea here just so the logger is accessible as Highlight::H.instance.logger or similar?

Looking at this again, I was hoping we could avoid some metaprogramming but I think it may be unavoidable.

I would say the only reasonable argument is to provide a way for users to handle opting out of logging (by simply not providing the logger param).